### PR TITLE
Update DeveloperGuide

### LINKED ISSUE #13: Update DG - Glossary

### PR DESCRIPTION
**Pull request by sachindavid96**
Closes #13. Update the Developer Guide with a glossary of key terms.

### CODE REVIEW

**Review by RubyNguyen07 — COMMENTED** [review]
Hmm I am not sure about whether we need to define project and deliverable, let's see what others think

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -398,8 +398,13 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Glossary
 
-* **Mainstream OS**: Windows, Linux, Unix, OS-X
-* **Private contact detail**: A contact detail that is not meant to be shared with others
+* **Posting**: An appointment to a software development job
+* **Console-first Interface**: A text-based user interface to efficiently
+                               run Mycelium
+* **Project**: A freelance software development job
+* **Contact**: An individual associated with a particular project
```
Comment (immanuelhume) [inline docs/DeveloperGuide.md:405]: I think we use the term "client" rather than contact right

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -398,8 +398,13 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* **Deliverable**: A task to be completed as part of a project
```
Comment (immanuelhume) [inline docs/DeveloperGuide.md:406]: did we use this word somewhere so far?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -398,8 +398,13 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Glossary
 
-* **Mainstream OS**: Windows, Linux, Unix, OS-X
-* **Private contact detail**: A contact detail that is not meant to be shared with others
+* **Posting**: An appointment to a software development job
+* **Console-first Interface**: A text-based user interface to efficiently
```
Comment (immanuelhume) [inline docs/DeveloperGuide.md:402]: I think this isn't too accurate. A "text-based" interface means something quite different from a GUI which is what we have. I think we can just define "console-first" as "an interface with interactions primarily through text commands"

**Review by immanuelhume — CHANGES_REQUESTED** [review]

**Comment by immanuelhume** [comment]
If there're not many term we can think of now, can just add 1 or 2 obvious one for now. Then add more as the weeks go by

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (sachindavid96) [reply docs/DeveloperGuide.md:406]: It was in one of the user stories

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (sachindavid96) [reply docs/DeveloperGuide.md:402]: That makes sense; will make the changes

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (sachindavid96) [reply docs/DeveloperGuide.md:405]: Sure

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (sachindavid96) [reply docs/DeveloperGuide.md]: On second thought, I think it is quite irrelevant; will remove it

**Review by sachindavid96 — COMMENTED** [review]
I have made the requested changes, in the glossary, to the definition of the console-first interface, changed 'contact' to 'client' and removed the definition of deliverables and posting

**Review by immanuelhume — APPROVED** [review]
LGTM :shipit:

**Pull request merged**